lib/test/index.js

```diff
@@ -284,9 +284,47 @@ describe('PropTypes', function () {
 
     parser.addType('NewsOrPhotos', _react.PropTypes.oneOf(['News', 'Photos']));
     var propTypesWithEnum = parser('{\n      mediaType: NewsOrPhotos\n    }');
+
     testPass(propTypesWithEnum.mediaType, undefined);
     testPass(propTypesWithEnum.mediaType, 'News');
     testPass(propTypesWithEnum.mediaType, 'Photos');
     testFail(propTypesWithEnum.mediaType, 'Video');
   });
+
+  it('should support union type.', function () {
+    var parser = (0, _index2.default)(_react.PropTypes);
+
+    var propTypes = parser('{\n      stringOrNumberOrNull: String | Number\n      stringOrNumber: String! | Number!\n      stringOrNumberGroup: (String | Number)!\n      stringOrNumberOrBoolean: (String | Number | Boolean)!\n      stringOrNumberOrBoolean2: ((String | Number)! | Boolean)!\n    }');
```

Comment:
Think this: `stringOrNumber: String | Number!`
Does this make sense? Is `null` be valid? Should parser warn user that this is probably not what it looks like?